python/lsst/summit/utils/guiders/reading.py

```diff
@@ -0,0 +1,724 @@
+    """
+    This class reads the stamp object from the Butler for one Guider and
+    converts them to DVCS view, making a new Stamps object
+
+    Parameters
+    ----------
+    detNum : `int`
+        Detector Id
+    seqNum : `int`
+        Sequence Number
+    dayObs : `int`
+        Day Observation
+    repo : `str`
+        Butler repo
+    collections : `list` of `str`
+        Butler collections
+    butler : `lsst.daf.butler.Butler`, optional
+        Butler object. If None, a new Butler will be created.
+    view : `str`, optional
+        View type, either 'dvcs' or 'ccd'. Default is 'dvcs'.
+    whichstamps : `list` of `int`, optional
+        List of stamp indices to read. If None, all stamps will be read.
+
+    Returns
+    -------
+    stamps : `lsst.meas.algorithms.stamps.Stamps`
+        Stamp images oriented in DVCS or CCD view, depending on the `view`
+        parameter.
+    """
+    # get Camera object
+    camera = LsstCam.getCamera()
+    detector = camera[detNum]
+
+    # for dayObs of 20250509 or before, the ROIs are swapped between SG0 and
+    # SG1. Fix here
+    ccd_swapped: str = ""
+    if dayObs < 20250509:
+        detName = camera[detNum].getName()
+        raft = detName[0:3]
+        ccd = detName[4:7]
+        if ccd == "SG0":
+            ccd_swapped = "SG1"
+        elif ccd == "SG1":
+            ccd_swapped = "SG0"
+
+        detName_swapped = raft + "_" + ccd_swapped
+        detector_swapped = camera[detName_swapped]
+        detNum_swapped = detector_swapped.getId()
+
+        dataId = {
+            "instrument": "LSSTCam",
+            "detector": detNum_swapped,
+            "day_obs": dayObs,
+            "seq_num": seqNum,
+        }
+    else:
+        dataId = {
+            "instrument": "LSSTCam",
+            "detector": detNum,
+            "day_obs": dayObs,
+            "seq_num": seqNum,
+        }
+
+    # finally read from the Butler
+    try:
+        raw_stamps = butler.get("guider_raw", dataId)
+    except DatasetNotFoundError:
+        raise StampsNotFoundError(
+            f"No guider data found for dayObs {dayObs}, seqNum {seqNum}, detector {detNum}."
+        )
+
+    md = raw_stamps.metadata
+
+    # fix CCD in the metadata
+    if dayObs < 20250509:
+        md["CCDSLOT"] = ccd_swapped
+
+    # also get the ampName
+    segment = md["ROISEG"]
+    ampName = "C" + segment[7:]
+
+    # build the CCD view Bounding Boxes
+    ccd_view_bbox = mk_roi_bboxes(md, camera)
+
+    # also build the Translation methods
+    # from CCD view -> DVCS view and the reverse
+    ft, bt = mk_ccd_to_dvcs(ccd_view_bbox, detector.getOrientation().getNQuarter())
+
+    if nstamps is None:
+        nstamps = md["N_STAMPS"]
+
+    timestamps = get_timestamps(raw_stamps, nstamps)  # to populate the metadata timestamps
+    # freq = timestamps.freq * 86400.0  # seconds
+
+    goodstamps = np.where(~timestamps.mask)[0].tolist()
+    badstamps = np.where(timestamps.mask)[0].tolist()
+    stampsDict = {}
```

Comment:
I'd add a comment about why some come out good/bad. I guess it's something bad in the data so that it doesn't parse, but would be good to explain why things aren't working.